transfuse-core/src/main/java/org/androidtransfuse/gen/FilerResourceWriter.java

```diff
@@ -41,18 +41,13 @@ public FilerResourceWriter(Filer filer) {
     public OutputStream openBinary(JPackage pkg, String fileName) throws IOException {
         FileObject resource = filer.createResource(StandardLocation.SOURCE_OUTPUT, pkg.name(), fileName);
 
-        OutputStream os = resource.openOutputStream();
-        openStreams.add(os);
-
+        OutputStream os = new WriterOutputStream(resource.openWriter());
         return os;
     }
 
 
     @Override
     public void close() throws IOException {
-        for (OutputStream openStream : openStreams) {
-            openStream.flush();
-            openStream.close();
-        }
```

Comment:
Why not keep these calls?  It seems like they would be necessary because we'll be dealing with a buffer with `WriterOutputStream`.

Reply:
it is causing this error
```
Caused by: java.io.IOException: Stream closed
    at sun.nio.cs.StreamEncoder.ensureOpen (StreamEncoder.java:45)
    at sun.nio.cs.StreamEncoder.flush (StreamEncoder.java:140)
    at java.io.OutputStreamWriter.flush (OutputStreamWriter.java:229)
    at java.io.FilterWriter.flush (FilterWriter.java:100)
    at org.androidtransfuse.util.apache.commons.WriterOutputStream.flush (WriterOutputStream.java:83)
    at org.androidtransfuse.gen.FilerSourceCodeWriter.close (FilerSourceCodeWriter.java:65)
    at com.sun.codemodel.JCodeModel.build (JCodeModel.java:312)
```


Seems it already closed by the WriterOutputStream
```
    public void close() throws IOException {
        this.processInput(true);
        this.flushOutput();
        this.writer.close();
    }
```
Do you have other clue why it causing error ?

Reply:
Ah, I bet codemodel is already closing the output stream and since it's closed flush throws an error.  You can outright get rid of the flush call instead of just commenting it out.

Reply:
yes it seems already flush and also closed from the log, also from the flamegraph.